app/src/main/java/org/fossasia/openevent/general/WelcomeFragment.kt

```diff
@@ -23,6 +37,53 @@ class WelcomeFragment : Fragment() {
         rootView.pickCityButton.setOnClickListener {
             Navigation.findNavController(rootView).navigate(R.id.searchLocationFragment, null, Utils.getAnimSlide())
         }
+
+        geoLocationViewModel.currentLocationVisibility.observe(this, Observer {
+            rootView.currentLocation.visibility = View.GONE
+        })
+
+        geoLocationViewModel.configure(null)
+
+        rootView.currentLocation.setOnClickListener {
+            checkLocationPermission()
+            geoLocationViewModel.configure(activity)
+            rootView.locationProgressBar.visibility = View.VISIBLE
+        }
+
+        geoLocationViewModel.location.observe(this, Observer { location ->
+            searchLocationViewModel.saveSearch(location)
+            redirectToMain()
+        })
+
         return rootView
     }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        when (requestCode) {
+            LOCATION_PERMISSION_REQUEST -> {
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    geoLocationViewModel.configure(activity)
+                } else {
+                    Snackbar.make(rootView, "Cannot fetch location!", Snackbar.LENGTH_SHORT).show()
+                    rootView.locationProgressBar.visibility = View.GONE
+                }
+            }
+        }
+    }
+
+    private fun checkLocationPermission() {
+        val permission =
+            ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.ACCESS_COARSE_LOCATION)
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION,
+                Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST)
+        }
+    }
+
+    private fun redirectToMain() {
+        val startMainActivity = Intent(activity, MainActivity::class.java).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
```

Comment:
It does not work in the same way as should. Activity is preparing again with slash theme.